### DISCUSSION TITLE
OpenAI LLM Client?

### DISCUSSION THREAD

**Discussion by nazargladish** [antiwork/shortest#218] [Ideas]
Would it be worth adding the OpenAI LLM client?

**Vision:**
Creating an abstract class that allows developers to inject their preferred client
This is apparently possible with current OpenAI functionality

**Comment by slavingia** [comment]
This is a bit too vague for me to understand what you mean. A user story / example would be helpful!

**Reply by nazargladish** [reply]
Perhaps overcomplicated language. What I basically meant is that we currently only support the Anthropic client, but I’m suggesting adding support for the OpenAI client as well

**Comment by slavingia** [comment]
My bad! Makes sense now. I agree once they have launched their version of computer use / operator. Apparently may be released in Jan.

**Comment by nazargladish** [comment]
This is indeed possible with current OpenAI capabilities. We can annotate a screenshot by processing the DOM, sending it to the LLM for analysis, and receiving the ID of the annotated element on the page. However, this approach would require a lot of new functionality and a bit of shift in our strategy for processing pages

Maybe worth waiting Computer Use feat from OpenAI

**Reply by slavingia** [reply]
Have you tried that? It doesn't really work in my experience.

**Comment by jiz17043** [comment]
Curious, currently only Anthropic API key is supported, will you consider supporting OpenAI APIkey?

**Reply by slavingia** [reply]
Once they launch a feature similar to computer use!
